File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/C/s010698290_commented.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    
    int penal[n+1];
    memset(penal, 0, sizeof(penal));
    
    char solved[n+1];
    memset(solved, 0, sizeof(solved));
    
    for(int i=0; i<m; i++) {
        int num;
        char str[2];
        scanf("%d %s", &num, str);
        if(str[0] == 'A') {
            solved[num] = 1;
        } else {
            if (solved[num] == 0) {
                penal[num]++;
            }
        }
    }
    
    int correct_ans = 0;
    int num_penal = 0;
    for(int i=1; i<=n; i++) {
        if(solved[i]) {
            correct_ans++;
            num_penal += penal[i];
        }
    }
    
    printf("%d %d\n", correct_ans, num_penal);
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Python/s010698290_commented.py
import sys

class FastReader:
    def __init__(self):
        self.br = sys.stdin
        self.st = None

    def next(self):
        while self.st == None or not self.st:
            try:
                self.st = self.br.readline().split()
            except Exception as e:
                print(e)
        return self.st.pop(0)

    def nextInt(self):
        return int(self.next())

    def nextLong(self):
        return int(self.next())

    def nextDouble(self):
        return float(self.next())

    def nextLine(self):
        try:
            return self.br.readline().strip()
        except Exception as e:
            print(e)

def main():
    s = FastReader()
    n = s.nextInt()
    m = s.nextInt()

    penal = [0] * (n + 1)
    solved = [False] * (n + 1)

    for _ in range(m):
        num = s.nextInt()
        str = s.next()
        if str[0] == 'A':
            solved[num] = True
        else:
            if not solved[num]:
                penal[num] += 1

    correct_ans = 0
    num_penal = 0
    for i in range(1, n + 1):
        if solved[i]:
            correct_ans += 1
            num_penal += penal[i]

    print(f"{correct_ans} {num_penal}")

if __name__ == "__main__":
    main()  # End of Code
